tensorflow/java/src/gen/cc/ops/op_generator.cc

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <string>
+
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/java/src/gen/cc/gen_util.h"
+#include "tensorflow/java/src/gen/cc/ops/op_generator.h"
+
+namespace tensorflow {
+
+OpGenerator::OpGenerator(const std::string& ops_lib)
+    : ops_lib(ops_lib) {
+
+  const std::string& lib_name = ops_lib.substr(0, ops_lib.find_last_of('_'));
+  package_name = gen_util::StripChar("org.tensorflow.op." + lib_name, '_');
+}
+
+OpGenerator::~OpGenerator() {}
+
+int OpGenerator::Run(const std::string& output_path) {
+  tensorflow::Env* env = tensorflow::Env::Default();
+  std::string package_path(output_path);
+  package_path += gen_util::ReplaceChar(package_name, '.', '/');
+
+  if (!env->FileExists(package_path).ok()) {
+    TF_CHECK_OK(env->RecursivelyCreateDir(package_path));
+  }
+
+  OpList ops;
+  OpRegistry::Global()->Export(true, &ops);
```

Comment:
Good, I'll pass the `OpList` as a argument to the `Run` method. I think it could make sense now to have the `ops_lib` name also as an argument to `Run` with the list of ops since they go together... (that may help you understand what is my mind set when I'm trying to partition what goes in the constructor and what doesn't ;) )